src/main/java/com/alistats/discorki/listener/SlashCommandListener.java

```diff
@@ -0,0 +1,200 @@
+
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+@Service
+public class SlashCommandListener extends ListenerAdapter {
+    @Autowired
+    private UserRepo userRepo;
+    @Autowired
+    private DiscordGuildRepo discordGuildRepo;
+    @Autowired
+    private RankRepo rankRepo;
+    @Autowired
+    private SummonerRepo summonerRepo;
+    @Autowired
+    private LeagueApiController leagueApiController;
+    @Autowired
+    private DiscordLeaderboardView discordLeaderboardView;
+
+    @Override
+    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
+        try {
+            switch (event.getName()) {
+                case "add":
+                    add(event);
+                    break;
+                case "remove":
+                    remove(event);
+                    break;
+                case "enable":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "disable":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "list_users":
+                    listUsers(event);
+                    break;
+                case "list_notifications":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "leaderboard":
+                    leaderboard(event);
+                    break;
+            }
+        } catch (Exception e) {
+            event.reply("An error occured.").queue();
+            e.printStackTrace();
+        }
+
+    }
+
+    private void add(SlashCommandInteractionEvent event) throws Exception {
+        // Get guild
+        DiscordGuild discordGuild = getOrCreateGuild(event.getGuild());
+        DiscordUser discordUser = new DiscordUser();
+
+        User user = event.getOption("user").getAsUser();
+        String summonerName = event.getOption("summoner_name").getAsString();
+
+        // Check if user is already in that guild
+        if (discordGuild.getUsers() != null
+                && discordGuild.getUsers().stream().anyMatch(u -> u.getId().equals(user.getId()))) {
+            // Get user
+            discordUser = discordGuild.getUsers().stream().filter(u -> u.getId().equals(user.getId())).findFirst()
+                    .get();
+        } else {
+            // Create user
+            discordUser.setId(user.getId());
+            discordUser.setUsername(user.getName());
+            discordUser.setDiscriminator(user.getDiscriminator());
+            discordUser.setGuild(discordGuild);
+            discordUser = userRepo.save(discordUser);
+        }
+
+        // Check if user already has that summoner
+        if (discordUser.getSummoners() != null
```

Comment:
Do these null checks have any added value? The response of getSummoners() will be an empty array if the list is empty and searching it will suffice. The response probably will never be null, as an Exception would be thrown.